Bluetooth: Classic: GOEP: Support multiple OBEX sessions

lylezhu2012 opened the pull request:
Split the structure `struct bt_obex` into two parts. 
Part 1 `struct bt_obex`, it is the common part and shared by all OBEX sessions which share the same one transport.
Part 2 `struct bt_obex_session`, it is the dedicated part for each OBEX session.

Add function `bt_obex_register()` to register the OBEX server with specific uuid.

Add function `bt_obex_unregister()` to unregister the OBEX server.

lylezhu2012 committed:
Bluetooth: Classic: GOEP: Support multiple OBEX sessions

Split the structure `struct bt_obex` into two parts.
Part 1 `struct bt_obex`, it is the common part and shared by all OBEX
sessions which share the same one transport.
Part 2 `struct bt_obex_server` and `struct bt_obex_client`. The
`struct bt_obex_server` is used to manage the OBEX session of server
role. The `struct bt_obex_client` is used to manage the OBEX session
of client role.

Add function `bt_obex_server_register()` to register the OBEX server
with specific uuid.

Add function `bt_obex_server_unregister()` to unregister the OBEX
server.

Add function `bt_obex_make_uuid()` to make UUID.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

MarkWangChinese commented on subsys/bluetooth/host/classic/obex.c:
```diff
@@ -1036,70 +1326,171 @@ int bt_obex_reg_transport(struct bt_obex *obex, const struct bt_obex_transport_o
 }
 
-int bt_obex_connect(struct bt_obex *obex, uint16_t mopl, struct net_buf *buf)
+int bt_obex_server_register(struct bt_obex_server *server, const struct bt_uuid_128 *uuid)
```
```suggestion
int bt_obex_server_register(struct bt_obex* obex, struct bt_obex_server *server, const struct bt_uuid_128 *uuid)
```
Then application don't needs to set `server->obex`. and highlight that the registering is for the `obex`.

lylezhu2012 commented on include/zephyr/bluetooth/classic/goep.h:
```diff
@@ -143,6 +143,11 @@ struct bt_goep_transport_rfcomm_server {
+	 *  Before returning the callback, @ref bt_goep::transport_ops should be initialized with
+	 *  valid address of type @ref bt_goep_transport_ops object. The field `mtu` of
+	 *  @ref bt_obex::rx could be passed with valid value. Or set it to zero, the mtu will be
+	 *  calculated according to @kconfig{CONFIG_BT_GOEP_L2CAP_MTU}.
```
Fixed.

lylezhu2012 commented on subsys/bluetooth/host/classic/obex.c:
```diff
@@ -75,12 +80,75 @@ static int obex_transport_disconn(struct bt_obex *obex)
+		break;
+	case BT_UUID_SIZE_32:
+		uuid->uuid.type = BT_UUID_TYPE_32;
+		uuid->u32.val = sys_get_be16(data);
```
Fixed.

lylezhu2012 commented on subsys/bluetooth/host/classic/obex.c:
```diff
@@ -99,19 +168,19 @@ static int obex_server_connect(struct bt_obex *obex, uint16_t len, struct net_bu
 	}
 
-	if (!obex->server_ops->connect) {
+	if (server->ops->connect == NULL) {
```
It is not needed. The `server->ops` has been checked in bt_obex_server_register().

lylezhu2012 committed:
Bluetooth: Classic: OBEX: Optimize MOPL configuration

Set the MOPL of RX and TX to `BT_OBEX_MIN_MTU` when registering OBEX
server.

Set the TX MOPL to `BT_OBEX_MIN_MTU` when sending OBEX connection
request.

Check if the MOPL of client exceeds MTU of transport when server
receives the connection request.

Check if the MOPL of server exceeds MTU of transport when client
receives the connection response.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>
